### PR TITLE
build: proper fix for the kerning issue in macOS build fixes  #16836

### LINKED ISSUE #16836: OSX: Bad kerning in disk image background
The background image generated by `make deploy` for our OSX disk image has bad kerning at least for the OSX version I tested.

Screen Shot 2019-09-09 at 12 57 33 PM

Software specs:
```
ProductName:    Mac OS X
ProductVersion: 10.14.6
BuildVersion:   18G87
```

### PR DESCRIPTION
**Pull request by RandyMcMillan**
After reading the librsvg documentation I realized that the svg settings were confusing the library.
In this case it is best to set view box dimensions.

The documentation states:

***If there are no width or height attributes in the toplevel <svg> -  librsvg looks for the viewBox attribute. If present, this defines the aspect ratio and a "natural" size in pixels.***

https://developer.gnome.org/rsvg/stable/ch02.html#id-1.2.4.2.6

LINE:4 Buy omitting the width and height attribs - we allow the librsvg to do its own interpretation in context.

LINE:26-32 By nesting the vector elements - we wrap the arrow into the text box. This keeps the arrow's coordinates relative to the text element.

LINE:29 Make the arrow grey to match the text.

Screen Shot 2019-10-27 at 9 27 20 AM

Thanks for your patience while I researched this and found a proper fix!

RM

### CODE REVIEW

File: contrib/macdeploy/background.svg [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
+  <defs>
+    <linearGradient id="gradient" x1="0%" y1="0%" x2="100%" y2="100%">
+      <stop offset="0%" style="stop-color:rgb(239,239,239);stop-opacity:1" />
+      <stop offset="33%" style="stop-color:rgb(239,239,239);stop-opacity:1" />
+      <stop offset="80%" style="stop-color:rgb(205,205,205);stop-opacity:1" />
+      <stop offset="100%" style="stop-color:rgb(204,204,204);stop-opacity:1" />
+    </linearGradient>
+  </defs>
+<rect width="1000" height="640" style="fill:url(#gradient);stroke-width:0" />
+  <g transform="translate(500,0) scale(1.0, 1.0)">
```
Comment (promag) [inline contrib/macdeploy/background.svg]: Why this change?

**Review by promag — COMMENTED** [review]
Should just replace width and height attributes with viewBox?

File: contrib/macdeploy/background.svg (hunk 1 above)
Comment (RandyMcMillan) [reply contrib/macdeploy/background.svg]: @promag - according to documentation the scaling/aspect ratio is taken care of in the 
preserveAspectRatio flag in 

LINE:4 
```html
<svg version="1.0" xmlns="http://www.w3.org/2000/svg" width="1000pt" height="640pt" viewBox="0 0 1000 640" preserveAspectRatio="xMidYMid meet">
```
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/preserveAspectRatio

```text
Meet or slice reference
The meet or slice reference is optional and, if provided, must be one of the following keywords:
meet (the default) - Scale the graphic such that:
aspect ratio is preserved
the entire viewBox is visible within the viewport
the viewBox is scaled up as much as possible, while still meeting the other criteria
```

Comment (RandyMcMillan) [reply contrib/macdeploy/background.svg]: Also https://developer.gnome.org/rsvg/stable/recommendations-assets.html

Comment (RandyMcMillan) [reply contrib/macdeploy/background.svg]: > Why this change?

Here is the result now with the previous scale settings
Screen Shot 2019-10-27 at 4 53 02 PM

Comment (RandyMcMillan) [reply contrib/macdeploy/background.svg]: @promag - now that I have a better understanding of how to configure this - it is now more of an aesthetic choice than dealing with a kerning issue. In fact I plan to rework the whole presentation in a later PR - I just want to fix the glaring issue now - then focus on a better presentation later.

**Review by promag — COMMENTED** [review]
Can you avoid the indentation change?

**Comment by laanwj** [comment]
> I just want to fix the glaring issue now

Yes, let's do the minimal change to fix the issue. This shouldn't need to go through multiple iterations and bikeshedding PRs. :smile:

**Comment by RandyMcMillan** [comment]
Part of the issue with this is the previous version of this file is that it isn't valid svg.

Anyway here is the result of the PR rebuilt and minimized diffs. 


Screen Shot 2019-10-28 at 1 06 42 PM

**Comment by RandyMcMillan** [comment]
@laanwj @promag - not sure what happened - but I will just leave this here. Be sure you have the correct commit.

**Comment by RandyMcMillan** [comment]
I pushed a diff minimized commit.

**Comment by laanwj** [comment]
Hmm, I think the whitespace lint is over-eager here, it should probably ignore image files.
